### PR TITLE
FINAP-107/price publishing controls

### PR DESCRIPTION
**Pull request by esuomi**
Simple publishing process and admin UI for taxi prices

Screenshot 2022-01-27 at 10 44 53

### CODE REVIEW

File: ote/src/clj/ote/services/taxiui_service.clj [hunk 1]
```diff
@@ -130,6 +131,19 @@
     (vec (->> (list-service-summaries db {:operator-ids groups})
               (map (fn [service] (update service :operating-areas #(db-util/PgArray->vec %))))))))
 
+(defn fetch-unapproved-prices
+  [db user]
+  ; TODO: check admin privileges for user
```
Comment (antti-nuortimo-solita) [inline ote/src/clj/ote/services/taxiui_service.clj:136]: Admin privilege seems to be missing from here.

File: ote/src/cljs/ote/app/controller/taxi_prices.cljs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            [ote.localization :refer [tr tr-key]]
+            [ote.communication :as comm]
+            [ote.app.controller.common :refer [->ServerError]]))
+
+(defrecord LoadTaxiPrices [])
+(defrecord LoadTaxiPricesResponse [response])
+(defrecord ApproveByIds [pricing-ids])
+(defrecord ApproveByIdsResponse [response])
+
+(def every-5min (* 1000 60 1))
```
Comment (antti-nuortimo-solita) [inline ote/src/cljs/ote/app/controller/taxi_prices.cljs:14]: I don't understand this. Why would `60000` be every 5 minutes? Do you mean (assuming time is in milliseconds) this:
`(def every-5min (* 1000 60 5))` ?

**Review by antti-nuortimo-solita — COMMENTED** [review]
Some issues that I have mentioned in the comments. Otherwise looks good.

File: ote/src/cljs/ote/app/controller/taxi_prices.cljs (hunk 2 above)
Comment (esuomi) [reply ote/src/cljs/ote/app/controller/taxi_prices.cljs:14]: Not my bug :-) Seems the place I copy-pasted this from has the same issue. Will fix.

File: ote/src/cljs/ote/views/admin/taxi_prices.cljs [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+                       :on-click #(do
+                                    (.preventDefault %)
+                                    (e! (admin-validation/->EditService service-id)))} name]]
+                 [ui/table-row-column (merge (stylefy/use-style style-base/table-col-style-wrap) {:width "8%"})
+                  (formatters/currency start-price-daytime)]
+                 [ui/table-row-column (merge (stylefy/use-style style-base/table-col-style-wrap) {:width "8%"})
+                  (formatters/currency start-price-nighttime)]
+                 [ui/table-row-column (merge (stylefy/use-style style-base/table-col-style-wrap) {:width "8%"})
+                  (formatters/currency start-price-weekend)]
+                 [ui/table-row-column (merge (stylefy/use-style style-base/table-col-style-wrap) {:width "10%"})
```
[the remark this answers is not in the document]
Comment (esuomi) [reply ote/src/cljs/ote/views/admin/taxi_prices.cljs:82]: Material UI table is a bit wonky, these number work, although they don't make much sense.

**Review by antti-nuortimo-solita — APPROVED** [review]
Looks good to me :ship:

**Pull request merged**